### PR TITLE
Replace ResolveAsyncValue in models

### PR DESCRIPTION
**Pull request by jrjohnson**
The TrackedAsyncData provides a better way to manage asynchronous state, however it is more verbose and in order to be auto tracked it requires each step in an async getter chain to be explicitly coded. I focused on the testable output of the non-private getters for this change and I think I've replaced our original resources with these new getters. Long term we probably need to remove many of the public getters on models because doing things this way where we can't know exactly where in an async chain a change may happen makes for some verbose and difficult to read code.

### CODE REVIEW

File: addon/models/learner-group.js [hunk 1]
```diff
@@ -192,44 +265,83 @@ export default class LearnerGroup extends Model {
...
     }
-    if (!this._parent?.allParents) {
-      return undefined;
+    const grandparents = await parent.getAllParents();
+
+    return [parent, ...grandparents];
+  }
+
+  @cached
+  get _grandparentsData() {
```
Comment (stopfstedt) [inline addon/models/learner-group.js:314]: suggesting name change to `_allAncestorsData()`.

**Review by stopfstedt — CHANGES_REQUESTED** [review]
Two minor suggestions for change, otherwise this looks fine to me (at least from what i can see with my bleeding eyes).

File: addon/models/learner-group.js [hunk 2]
```diff
@@ -107,15 +163,26 @@ export default class LearnerGroup extends Model {
     return ++offset;
   }
 
+  @cached
+  get _grandChildrenData() {
```
[the remark this answers is not in the document]
Comment (jrjohnson) [reply addon/models/learner-group.js]: ✅

File: addon/models/learner-group.js (hunk 1 above)
Comment (jrjohnson) [reply addon/models/learner-group.js]: ✅

**Review by stopfstedt — APPROVED** [review]

**Pull request merged**